app/src/main/java/com/futurice/freesound/feature/home/HomeFragmentViewModel2Factory.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017 Futurice GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.futurice.freesound.feature.home;
+
+import com.futurice.freesound.feature.common.scheduling.SchedulerProvider;
+
+import android.arch.lifecycle.ViewModel;
+import android.arch.lifecycle.ViewModelProvider;
+
+class HomeFragmentViewModel2Factory implements ViewModelProvider.Factory {
```

Comment:
I found a good way of doing that, but I might have to juggle it around at bit because you can't call reified functions from Java where this factory is called from.